pkgs/development/python-modules/pproxy/default.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, pythonOlder
+, buildPythonPackage
+, fetchFromGitHub
+, pycryptodome
+, uvloop
+}:
+
+buildPythonPackage rec {
+  pname = "pproxy";
+  version = "2.3.2";
+
+  disabled = pythonOlder "3";
+
+  # doesn't use tagged releases. Tests not in PyPi versioned releases
+  src = fetchFromGitHub {
+    owner = "qwj";
+    repo = "python-proxy";
+    rev = "818ab9cc10565789fe429a7be50ddefb9c583781";
+    sha256 = "0g3cyi5lzakhs5p3fpwywbl8jpapnr8890zw9w45dqg8k0svc1fi";
+  };
+
+  propagatedBuildInputs = [
+    pycryptodome
+    uvloop
+  ];
+
+  doCheck = true;
+
+  pythonImportsCheck = [ "pproxy" ];
+  disabledTests = [ "api_server" "api_client" ];  # try to connect to outside Internet, so disabled
+  # test suite doesn't use test runner. so need to run ``python ./tests/*``
+  checkPhase = ''
+    shopt -s extglob
+    for f in /build/source/tests/!(${builtins.concatStringsSep "|" disabledTests}).py ; do
+      echo "***Testing $f***"
+      eval "python $f"
+    done
+  '';
```

Comment:
should be:
```
  checkInputs = [ pytest ];
  checkPhase = ''
    pytest --ignore tests/api_server.py --ignore tests/api_client.py
  '';
```

Reply:
I spent about 10 mins trying to get these tests to work with pytest, and I don't think it will work. they're just not designed for a test suite runner at all, and are only meant to be run with ``python $FILE``, due to function naming, requiring command line arguments, and having no ``if __name__ == "__main__"`` guards. Full notes below.

Reply:
When I run above, it doesn't collect or run any tests.

When I run ``pytest --ignore[...] ./tests/*.py``, it actually tries to run the tests, but doesn't ignore the proper tests. ``pytest ./tests/cipher_*.py`` works slightly better

Issue is that tests are not guarded by ``if __name__ == "__main__"`` or meant for any test runner (pytest/nose/tox), and that they include code similar to ``cipher = sys.argv[1] if len(sys.argv) > 1 else None``, which grabs ``--ignore=...``